unit-tests/src/test/java/org/eclipse/collections/impl/utility/IterateTest.java

```diff
@@ -739,6 +752,97 @@ public void toMapSelectingKeyAndValue()
     }
 
     @Test
+    public void toMultimap()
+    {
+        MutableList<Integer> list = Lists.mutable.of(1, 2, 2, 3, 3, 3);
+
+        Pair[] expectedPairs = {Tuples.pair(1, "1"),
+                Tuples.pair(1, "1"),
+                Tuples.pair(2, "1"),
+                Tuples.pair(2, "2"),
+                Tuples.pair(2, "1"),
+                Tuples.pair(2, "2"),
+                Tuples.pair(3, "1"),
+                Tuples.pair(3, "3"),
+                Tuples.pair(3, "1"),
+                Tuples.pair(3, "3"),
+                Tuples.pair(3, "1"),
+                Tuples.pair(3, "3")};
+
+        Function<Integer, Integer> keyFunction = each -> each;
+        Function<Integer, Iterable<String>> valuesFunction = each -> Lists.mutable.of("1", String.valueOf(each));
+        MutableMultimap<Integer, String> actualMultimap1 = Iterate.toMultimap(list, keyFunction, valuesFunction, Multimaps.mutable.list.empty());
+        Verify.assertListMultimapsEqual(FastListMultimap.newMultimap(expectedPairs), (ListMultimap<Integer, String>) actualMultimap1);
```

Comment:
You can get rid of the cast to `ListMultimap` if you change the assignment on the previous line to type `MutableListMultimap`. I think all casts in this test are avoidable.